R/060_utils.R

```diff
@@ -0,0 +1,348 @@
+    filled_join <-
+      header_df %>%
+      add_v_border_groups(formats) %>%
+      dplyr::group_by(v_border_group) %>%
+      dplyr::select(row, col, v_border_group, character) %>%
+      dplyr::mutate(value = ifelse(is.na(character), paste3(character, collapse = " _ ") %>% stringr::str_remove_all(" _ "), character)) %>%
+      dplyr::ungroup() %>%
+      dplyr::select(row, col, character = value)
+    
+    header_df <-
+      header_df %>%
+      dplyr::select(-character) %>%
+      dplyr::left_join(filled_join, by = c("row", "col")) %>%
+      dplyr::arrange(row, col)
+  }
+  
+  header_df
+}
+
+#' Unmerge and fill in blanks for col headers.
+#'
+#' This is an internal function that function ensures that merged cells are unmerged.
+#'
+#' @param header_df a data frame containing header cells.
+#' @param header_fill the criteria according to which blank cells will be filled.
+#' @param formats the formats of the workbook associated with the tidyxl data frame.
+
+fill_blanks_in_col_headers <- function(header_df, header_fill, formats) {
+  if (header_fill == "style") {
+    continue <- TRUE
+    
+    while (continue) {
+      sheet_original <- header_df
+      header_df <- header_df %>% unmerge_cells(strict_merging = FALSE)
+      
+      continue <- !identical(sheet_original, header_df)
+    }
+  }
+  if (header_fill == "local_format_id") {
+    continue <- TRUE
+    
+    while (continue) {
+      sheet_original <- header_df
+      header_df <- header_df %>% unmerge_cells(strict_merging = TRUE)
+      
+      continue <- !identical(sheet_original, header_df)
+    }
+  }
+  
+  if (header_fill == "borders") {
+    filled_join <-
+      header_df %>%
+      add_h_border_groups(formats) %>%
+      dplyr::group_by(h_border_group) %>%
+      dplyr::select(row, col, h_border_group, character) %>%
+      dplyr::mutate(value = ifelse(is.na(character), paste3(character, collapse = " _ ") %>%
+                                     stringr::str_remove_all(" _ "), character)) %>%
+      dplyr::ungroup() %>%
+      dplyr::arrange(h_border_group, row, col) %>%
+      dplyr::select(row, col, character = value)
+    
+    header_df <-
+      header_df %>%
+      dplyr::select(-character) %>%
+      dplyr::left_join(filled_join, by = c("row", "col"))
+  }
+  
+  header_df
+}
+
+
+
+#' Unmerge and fill in blanks for headers
+#'
+#' This is an internal function that function ensures that merged cells are unmerged.
+#'
+#' @param header_df a data frame containing header cells.
+#' @param header_fill he criteria according to which blank cells will be filled.
+#' @param formats the formats of the workbook associated with the tidyxl data frame.
+#' @param direction the direction in which headers are filled.
+
+fill_blanks_in_headers <- function(header_df, header_fill, formats, direction) {
+  if (direction %in% c("N", "S", "up", "down")) {
+    header_df <- suppressMessages(fill_blanks_in_col_headers(header_df, header_fill, formats))
```

Comment:
Why `suppressMessages`?  Is that because #30 was noisy?

Reply:
From memory, dplyr::left_join was producing a message for each iteration for the while loop in fill_header_blanks. I'll have a second look at this.

Reply:
Addressed in 6f6a5bcd0b6f80080eb9cb32962494fc5d62173e.